gradle/package-info.gradle

```diff
@@ -0,0 +1,65 @@
+import java.nio.file.Files
+
+task generateImplPackageInfos(type: GenerateImplPackageInfos) {
+	group = 'fabric'
+	description = 'Generates package-info files for implementation packages.'
+
+	for (def sourceSet in [sourceSets.main, sourceSets.client]) {
+		sourceRoots.from sourceSet.java.srcDirs
+	}
+
+	header = rootProject.file('HEADER')
+	outputDir = file('src/generated/java')
+}
+
+sourceSets.main.java.srcDir generateImplPackageInfos.outputDir
```

Comment:
I think this might need splitting for the main/client source sets as its not going to be correctly denoting the package-infos for client only packages. Thus partly breaking splitting the jars in a mod dev env.